openquake/hazardlib/shakemap/conditioned_gmfs.py

```diff
@@ -0,0 +1,988 @@
+                    for i, gmv in enumerate(gmfa):
+                        if gmv.sum() == 0:
+                            continue
+                        data["sid"].append(sids[i])
+                        data["eid"].append(eid)
+                        data["rlz"].append(rlz)  # used in compute_gmfs_curves
+                        for m in range(M):
+                            data[f"gmv_{m}"].append(gmv[m])
+                        for outkey, outarr in items:
+                            data[outkey].append(outarr[i])
+                        # gmv can be zero due to the minimum_intensity, coming
+                        # from the job.ini or from the vulnerability functions
+                n += len(eids)
+        return data
+
+    def compute(self, gsim, num_events, mean_covs, rng):
+        """
+        :param gsim: GSIM used to compute mean_stds
+        :param num_events: the number of seismic events
+        :param mean_covs: array of shape (4, M, N)
+        :returns:
+            a 32 bit array of shape (num_imts, num_sites, num_events) and
+            two arrays with shape (num_imts, num_events): sig for tau
+            and eps for the random part
+        """
+        M = len(self.imts)
+        num_sids = mean_covs[0][self.imts[0].string].size
+        result = numpy.zeros((M, num_sids, num_events), F32)
+        sig = numpy.zeros((M, num_events), F32)  # same for all events
+        eps = numpy.zeros((M, num_events), F32)  # not the same
+        numpy.random.seed(self.seed)
+
+        for m, imt in enumerate(self.imts):
+            mu_Y_yD = mean_covs[0][imt.string]
+            # cov_Y_Y_yD = mean_covs[1][imt.string]
+            cov_WY_WY_wD = mean_covs[2][imt.string]
+            cov_BY_BY_yD = mean_covs[3][imt.string]
+            try:
+                result[m], sig[m], eps[m] = self._compute(
+                    mu_Y_yD, cov_WY_WY_wD, cov_BY_BY_yD, imt, num_events, rng
+                )
+            except Exception as exc:
+                raise RuntimeError(
+                    "(%s, %s, source_id=%r) %s: %s"
+                    % (gsim, imt, decode(self.source_id), exc.__class__.__name__, exc)
+                ).with_traceback(exc.__traceback__)
+        if self.amplifier:
+            self.amplifier.amplify_gmfs(self.ctx.ampcode, result, self.imts, self.seed)
+        return result, sig, eps
+
+    def _compute(self, mu_Y, cov_WY_WY, cov_BY_BY, imt, num_events, rng):
+        if self.cmaker.truncation_level == 0:
+            gmf = exp(mu_Y, imt)
+            gmf = gmf.repeat(num_events, axis=1)
+            inter_sig = 0
+            inter_eps = [numpy.zeros(num_events)]
+        else:
+            cov_Y_Y = cov_WY_WY + cov_BY_BY
+            # cov_Y_Y_near = cov_nearest(
+            #     cov_Y_Y, method="clipped", threshold=1e-15, n_fact=100, return_all=False
+            # )
+            # cov_Y_Y_near = numpy.nan_to_num(cov_Y_Y_near).clip(min=0)
```

Comment:
what's the purpose of the commented lines? should we remove them?

Reply:
Removed